tests/scale/scale_test.go

```diff
@@ -0,0 +1,254 @@
+//go:build scale
+// +build scale
+
+package scale
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// testing flags
+var (
+	numIterations = flag.Int("i", 1, "number of times to scale the resource")
+	delay         = flag.Duration("delay", 0, "delay between each scaling iteration")
+	version       = flag.String("version", "1.0", "version of NGF under test")
+)
+
+func TestScale_Listeners(t *testing.T) {
+	ip := getIP(t)
+	url := fmt.Sprintf("http://%s/", ip)
+
+	runScaleTest(
+		t,
+		[]string{"# Listeners", "Time to Ready (s)", "Error"},
+		func(dir string) error {
+			return generateScaleListenerManifests(*numIterations, dir, false /*non-tls*/)
+		},
+		url,
+	)
+}
+
+func TestScale_HTTPSListeners(t *testing.T) {
+	ip := getIP(t)
+	url := fmt.Sprintf("https://%s/", ip)
+
+	runScaleTest(
+		t,
+		[]string{"# HTTPS Listeners", "Time to Ready (s)", "Error"},
+		func(dir string) error {
+			return generateScaleListenerManifests(*numIterations, dir, true /*tls*/)
+		},
+		url,
+	)
+}
+
+func TestScale_HTTPRoutes(t *testing.T) {
+	ip := getIP(t)
+	url := fmt.Sprintf("http://%s/", ip)
+
+	runScaleTest(
+		t,
+		[]string{"# HTTPRoutes", "Time to Ready (s)", "Error"},
+		func(dir string) error {
+			return generateScaleHTTPRouteManifests(*numIterations, dir)
+		},
+		url,
+	)
+}
+
+func runScaleTest(
+	t *testing.T,
+	resultHeaders []string,
+	generateManifests func(dir string) error,
+	url string,
+) {
+	t.Helper()
+	manifestDir := t.Name()
+
+	writer := newResultsWriter(t, t.Name(), resultHeaders...)
+
+	if err := generateManifests(manifestDir); err != nil {
+		t.Fatalf("failed to generate manifests: %s", err)
+	}
+
+	startTime := time.Now()
+	startUnix := fmt.Sprintf("%d", startTime.Unix())
+
+	if err := kubectlApply(getPrereqDirName(manifestDir)); err != nil {
+		t.Fatalf("failed to apply prerequisite resources: %s", err)
+	}
+
+	t.Log("Waiting for all Pods to be Ready")
+	if err := kubectlWaitAllPodsReady(); err != nil {
+		t.Fatalf("failed to wait for all Pods to be Ready: %s", err)
+	}
+
+	for i := 0; i < *numIterations; i++ {
+		t.Logf("Scaling up to %d resources", i)
+
+		manifestFile := filepath.Join(manifestDir, fmt.Sprintf("manifest-%d.yaml", i))
+
+		if err := kubectlApply(manifestFile); err != nil {
+			t.Errorf("failed to scale up: %s", err)
+		}
+
+		host := fmt.Sprintf("%d.example.com", i)
+
+		t.Logf("Sending request to url %s with host %s...", url, host)
+
+		ttr, err := waitForResponseForHost(url, host)
+
+		seconds := ttr.Seconds()
+		record := []string{strconv.Itoa(i + 1), strconv.FormatFloat(seconds, 'f', -1, 64)}
+		if err != nil {
+			record = append(record, err.Error())
+		}
+
+		err = writer.Write(record)
+		if err != nil {
```

Comment:
What's the reason behind not using inline error checking here?

Reply:
Just missed it. I'll fix